source/extensions/filters/http/oauth/config.cc

```diff
@@ -0,0 +1,86 @@
+#include "extensions/filters/http/oauth/config.h"
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include "envoy/api/v2/auth/secret.pb.h"
+#include "envoy/common/exception.h"
+#include "envoy/extensions/filters/http/oauth/v3/oauth.pb.validate.h"
+#include "envoy/registry/registry.h"
+#include "envoy/secret/secret_manager.h"
+#include "envoy/secret/secret_provider.h"
+#include "envoy/ssl/private_key/private_key.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/common/assert.h"
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/http/oauth/filter.h"
+#include "extensions/filters/http/oauth/oauth.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Oauth {
+
+namespace {
+Secret::GenericSecretConfigProviderSharedPtr
+secretsProvider(const envoy::extensions::transport_sockets::tls::v3::SdsSecretConfig& config,
+                Secret::SecretManager& secret_manager,
+                Server::Configuration::TransportSocketFactoryContext& transport_socket_factory) {
+  if (config.has_sds_config()) {
+    return secret_manager.findOrCreateGenericSecretProvider(config.sds_config(), config.name(),
```

Comment:
How does warming work here? The listener warms on SDS for its own secrets, but if a listener has an HCM with an OAuth filter that needs some SDS secrets, how does is this warming dependency implemented?

Reply:
You raise a good point, I don't think this would work nicely right now: I think we'd be seeing a null secret until the secret has been retrieved, so we'd fail requests until that happens. 

Do we have any mechanism like the init manager to add several initialization steps to the listener? I imagine we should block listener warming until all the required secrets have been fetched.

Reply:
Another option would be to only allow static secrets, though it would be nice to have full SDS support

Reply:
Yes, the init manager can support having multiple dependencies, that's how a listener can warm on RDS and SDS today. You can also have transitive init manager as done in SRDS. Since you should have the filter config in HCM, which should be inside the Listener object, then you should be able to do warming dependencies using the listener's init manager.

@kyessenov how is this going to work with ExtensionConfigDS? I.e. you have a listener, that then depends on ExtensionConfigDS for an OAuth filter, which then depends on SDS. Seems it might be necessary to have a transitive init manager in that case.

Reply:
Do filters delay initialization today? git grep initManager -- source/extensions/filters/ shows nothing. I imagine you are right, we do need a separate init manager in HCM  and another one in FilterConfigProviderManager.

Reply:
Digging through the SDS code it seems like it does use the init manager provided by `TransportSocketFactoryContext` to presumably delay warming:  this seems to be `dynamic_init_manager_` from the `ListenerImpl`, so maybe this does the right thing?